### PR TITLE
[DNM v2.5.9/v2.6] Add additional config options for fluentd

### PR DESCRIPTION
**Pull request by dbason**
This change should meet the requirements for https://github.com/rancher/rancher/issues/31807

The replicas setting may be one worth plumbing through to the UI, but I haven't done this yet.

For tolerations and nodeSelectors, fluentd will use the chart wide settings unless overridden with fluent specific settings.

### CODE REVIEW

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/aks/logging.yaml
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/aks/logging.yaml:35]: This should be wrapped to ensured that replicas are populated

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/eks/logging.yaml
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/eks/logging.yaml:36]: Same as above

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/gke/logging.yaml
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/gke/logging.yaml:35]: Same as above

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/k3s/logging-k3s-openrc.yaml
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/k3s/logging-k3s-openrc.yaml:42]: Same as above

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/k3s/logging-k3s-systemd.yaml
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/k3s/logging-k3s-systemd.yaml:42]: Same as above

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/rke2/logging-rke2-containers.yaml
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/rke2/logging-rke2-containers.yaml:47]: Same as above

File: packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml
Comment (nickgerace) [inline packages/rancher-logging/generated-changes/overlay/templates/loggings/root/logging.yaml:64]: Same as above

**Review by nickgerace — CHANGES_REQUESTED** [review]

**Comment by nickgerace** [comment]
Thanks for the PR btw @dbason :)
I have the same question as @paynejacob did, and made one comment that exists in the Logging CRs.
After those are addressed, this will more than likely be good to go!

File: packages/rancher-logging/generated-changes/patch/values.yaml.patch
Comment (dbason) [reply packages/rancher-logging/generated-changes/patch/values.yaml.patch:87]: If the scaling is null in the Fluentd spec the operator creates it and populates it with 1 replica:
https://github.com/banzaicloud/logging-operator/blob/e53255c84b2d4e1f86f6ac628cc080d716ca9fa9/pkg/sdk/api/v1beta1/logging_types.go#L233-L237

**Comment by dbason** [comment]
@nickgerace @paynejacob I've updated the commit to address the feedback.  I've also added the bufferStorageVolume in response to a request on the original issue.

**Review by nickgerace — DISMISSED** [review]
LGTM after squashing all commits and rebase. Like the `disablePvc` check too before allowing the `bufferStorageVolume` too

**Comment by dbason** [comment]
I already rebased to fix the failing CI job, I'll squath the second commit in now.

**Review by nickgerace — APPROVED** [review]

**Comment by nickgerace** [comment]
@dbason would you mind rebasing one more time?

**Comment by nickgerace** [comment]
Relevant issues:
- https://github.com/rancher/rancher/issues/31807
- https://github.com/rancher/rancher/issues/28401

File: packages/rancher-logging/package.yaml
Comment (nickgerace) [inline packages/rancher-logging/package.yaml:3]: This should actually be
```
packageVersion: 01
releaseCandidateVersion: 00
```
since this is a v2.6/v2.5.9 chart

**Review by nickgerace — CHANGES_REQUESTED** [review]

File: packages/rancher-logging/package.yaml
Comment (dbason) [reply packages/rancher-logging/package.yaml]: > @dbason would you mind rebasing one more time?

@nickgerace done, and the version updated :)

**Comment by dbason** [comment]
> @dbason would you mind rebasing one more time?

Done, and the version is updated :)

**Review by nickgerace — APPROVED** [review]
LGTM. Let's wait for v2.5.8 release to merge

**Comment by nickgerace** [comment]
I believe we are good to close this since https://github.com/rancher/charts/pull/1182 has been merged with @dbason as a co-committer

**Pull request closed without merging**
